packages/devtools/lib/src/timeline/cpu_profile_protocol.dart

```diff
@@ -122,7 +122,7 @@ class CpuProfileData {
   void _setExclusiveSampleCounts() {
     for (Map<String, dynamic> traceEvent in stackTraceEvents) {
       final leafId = traceEvent[stackFrameId];
-      stackFrames[leafId].exclusiveSampleCount++;
+      stackFrames[leafId]?.exclusiveSampleCount++;
```

Comment:
why do we have a case where the leafId is missing?
Can we add a test case that fails without this check and succeeds with the check?
I'm worried this change means we are quietly dropping frames if we don't find the expected ID.
Is it possible we are sometimes getting data out of order in a way that causes this issue?

Reply:
I asked ben about it, but I haven't heard back. It's not an ordering issue in this case - the stack frame just isn't in the response when it should be.

I agree we don't want to be silently dropping frames, but this check is to protect ourselves against bad data so it is necessary for now at least. Hopefully Ben will be able to help us identify the root cause and we can revert this check.

Reply:
I added a test. I was having a hard time repro-ing so I added an assert with instructions to export the timeline and send it to me if anyone else hits this while working on DevTools.